api/src/main/java/io/strimzi/api/kafka/model/balancing/KafkaRebalanceMode.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.balancing;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum KafkaRebalanceMode {
+    FULL_REBALANCE,
+    ADD_BROKER,
+    REMOVE_BROKER;
+
+    @JsonCreator
+    public static KafkaRebalanceMode forValue(String value) {
+        switch (value) {
+            case "full-rebalance":
+                return FULL_REBALANCE;
+            case "add-broker":
+                return ADD_BROKER;
+            case "remove-broker":
+                return REMOVE_BROKER;
+            default:
+                return null;
+        }
+    }
+
+    @JsonValue
+    public String toValue() {
+        switch (this) {
```

Comment:
You could avoid needing this switch statement (and the one above) by making the string version of the enum a field of the enum class itself (`name` for example) and passing it as an argument to the constructor. Then your `toValue` method would just be `return this.name` or similar?

eg
```java
FULL_REBALANCE ("full-rebalance"),
ADD_BROKER ("add-broker"),
REMOVE_BROKER ("remove-broker");

private String name;
public KafkaRebalanceMode(String name){
    this.name = name
}

public String toValue() {
    return this.name;
}
```

Reply:
Yeah, much better.